parser/unparser.go

```diff
@@ -156,24 +156,18 @@ func (un *unparser) visitCallBinary(expr *exprpb.Expr) error {
 func (un *unparser) visitCallConditional(expr *exprpb.Expr) error {
 	c := expr.GetCallExpr()
 	args := c.GetArgs()
-	// add parens if operand is a conditional itself.
-	nested := isSamePrecedence(operators.Conditional, args[0])
-	err := un.visitMaybeNested(args[0], nested)
+	err := un.visitMaybeParenExpr(args[0])
```

Comment:
I don't believe this change is strictly necessary since the conditional is the highest precedence operator and so the following expressions are equivalent:

`(x > 5) ? (x - 5) : 0`  -> `x > 5 ? x - 5 : 0`

I would leave the conditional unmodified, but add a new test case to `TestUnparse_Equivalent`

Reply:
Yes, I just ran into that myself testing further. I have now changed it such that it combines both, in order to properly wrap conditionals, but at the same time also provide the slightly more readable inner expressions in parenthesis if applicable.